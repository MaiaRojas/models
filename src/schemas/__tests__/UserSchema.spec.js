const mongoose = require('mongoose/browser');
const { UserSchema } = require('../')(mongoose);

describe('UserSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, UserSchema);
    return doc.validate()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should pass validation when at least uid and email', () => {
    const doc = new mongoose.Document({
      uid: 'xxxxxxxxxx',
      email: 'foo@bar.baz',
    }, UserSchema);
    return doc.validate();
  });
});
