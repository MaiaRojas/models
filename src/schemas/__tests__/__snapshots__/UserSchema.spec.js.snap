// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSchema should fail validation when fields missing 1`] = `
Object {
  "email": [ValidatorError: Path \`email\` is required.],
  "uid": [ValidatorError: Path \`uid\` is required.],
}
`;
